src/reconcile/rec_visibility.c

```diff
@@ -756,7 +756,8 @@ __rec_upd_select(WT_SESSION_IMPL *session, WTI_RECONCILE *r, WT_UPDATE *first_up
         /*
          * Special handling for application threads evicting their own updates.
          */
-        if (!is_hs_page && F_ISSET(r, WT_REC_APP_EVICTION_SNAPSHOT) && txnid == session_txnid) {
+        if (!is_hs_page && F_ISSET(r, WT_REC_APP_EVICTION_SNAPSHOT) &&
+          session_txnid != WT_TXN_NONE && txnid == session_txnid) {
```

Comment:
In Disagg, the durability guarantees for Eviction changes.
1. Eviction should not evict a unstable update (timestamp < stable_timestamp)
2. Should you checkpoint pinned id if checkpoint is in progress, else latest txn snapshot.

In Disagg the eviction visibility should be a earlier that the checkpoint in which the evicted page will be part of.